src/Relude/Extra/Foldable1.hs

```diff
@@ -139,6 +155,22 @@ instance Foldable1 NonEmpty where
     {-# INLINE maximum1 #-}
     {-# INLINE minimum1 #-}
 
+    maximumOn1 :: Ord b => (a -> b) -> NonEmpty a -> a
+    maximumOn1 func = foldl1' (cmpOn func)
+      where
+        cmpOn p a b = case p a `compare` p b of
+                        GT -> a
+                        _  -> b
```

Comment:
The alignment here doesn't match our style guide. It looks like currently it depends on the length of the identifier on a previous line, but always should be the same. Like this:

```haskell
cmpOn ...
    GT -> ...
    _ ->
```